aspnetcore/host-and-deploy/iis/troubleshoot.md

```diff
@@ -15,46 +15,138 @@ uid: host-and-deploy/iis/troubleshoot
+![Browser window showing the 502.5 Process Failure page](troubleshoot/_static/process-failure-page.png)
 
-When Kestrel starts normally behind IIS but the app won't run on the system after successfully running locally, an environment variable can be temporarily added to *web.config* to set the `ASPNETCORE_ENVIRONMENT` to `Development`. As long as the environment isn't overridden in app startup, setting the environment variable allows the [developer exception page](xref:fundamentals/error-handling) to appear when the app is run. Setting the environment variable for `ASPNETCORE_ENVIRONMENT` in this way is only recommended for staging/testing servers that aren't exposed to the Internet. Be sure to remove the environment variable from the *web.config* file when finished. For information on setting environment variables via *web.config*, see [environmentVariables child element of aspNetCore](xref:host-and-deploy/aspnet-core-module#setting-environment-variables).
+**500 Internal Server Error**  
+The app starts, but an error prevents the server from fulfilling the request.
 
-In most cases, enabling application logging assists in troubleshooting problems with the app or the reverse proxy. See [Logging](xref:fundamentals/logging/index) for more information.
+This error occurs within the app's code during startup or while creating a response. The response may contain no content, or the response may appear as a *500 Internal Server Error* in the browser. The Application Event Log usually states that the app started normally. From the server's perspective, that's correct. The app did start, but it can't generate a valid response. [Run the app at a command prompt](#run-the-app-at-a-command-prompt) on the server or [enable the ASP.NET Core Module stdout log](#aspnet-core-module-stdout-log) to troubleshoot the problem.
 
-The last troubleshooting tip pertains to apps that fail to run after upgrading either the .NET Core SDK on the development machine or package versions within the app. In some cases, incoherent packages may break an app when performing major upgrades. Most of these issues can be fixed by:
+## Default startup limits
 
-* Deleting the `bin` and `obj` folders in the project.
-* Clearing package caches at `%UserProfile%\.nuget\packages\` and `%LocalAppData%\Nuget\v3-cache`.
-* Restoring and rebuilding the project.
-* Confirming that the prior deployment on the server has been completely deleted prior to re-deploying the app.
+The ASP.NET Core Module is configured with a default *startupTimeLimit* (120 seconds) and *startupRetryCount* (2). A slow-starting app may take up to six minutes before the module logs a process failure. For information on configuring these settings, see [Attributes of the aspNetCore element](xref:host-and-deploy/aspnet-core-module#attributes-of-the-aspnetcore-element).
```

Comment:
"A slow-starting app may take up to six minutes" ??

Reply:
120 seconds (2 minutes) * (initial attempt + 2 retries) ... doesn't that come out to a six minute period until the ANCM finally gives up?

Reply:
> A slow-starting app may take .....

That wording kind'a sucks tho. lol

How about ...

> An app may take .......

Reply:
What is `startupRetryCount`?

Reply:
*I made that up!* :smile: It was accidently held over from HttpPlatformHandler days. https://docs.microsoft.com/en-us/iis/extensions/httpplatformhandler/httpplatformhandler-configuration-reference

No worries ... I'll take it out. The startup time limit is 120 seconds.